In [4]:
import pandas as pd
import glob
import os

## 대민 데이터 결측치 / 이상치

In [5]:
file_path = r"C:\Users\cute3\OneDrive\Desktop\pro\final_project\Data\신용정보.csv"

# 파일 존재 여부 먼저 확인
print(os.path.exists(file_path))

# True가 나오면 실행
df_credit = pd.read_csv(file_path, encoding="utf-8-sig")

df_credit.head(20)

True


,BS_YR_MON,SIGUNGU,BCD,AGE,TOT_CNT,ECON_CNT,NECON_CNT,읍면동명,시군구명
0,202301,41131,4113110100,20,2984,2874,66,신흥1동,성남시 수정구
1,202301,41131,4113110100,20,2984,2874,66,신흥2동,성남시 수정구
2,202301,41131,4113110100,20,2984,2874,66,신흥3동,성남시 수정구
3,202301,41131,4113110100,30,3523,3375,105,신흥1동,성남시 수정구
4,202301,41131,4113110100,30,3523,3375,105,신흥2동,성남시 수정구
5,202301,41131,4113110100,30,3523,3375,105,신흥3동,성남시 수정구
6,202301,41131,4113110100,40,3467,3266,156,신흥1동,성남시 수정구
7,202301,41131,4113110100,40,3467,3266,156,신흥2동,성남시 수정구
8,202301,41131,4113110100,40,3467,3266,156,신흥3동,성남시 수정구
9,202301,41131,4113110100,50,4071,3656,318,신흥1동,성남시 수정구


In [3]:
df_credit.shape

(29818, 9)

In [4]:
# 결측치 보기
df_credit.isnull().sum()

BS_YR_MON    0
SIGUNGU      0
BCD          0
AGE          0
TOT_CNT      0
ECON_CNT     0
NECON_CNT    0
읍면동명         0
시군구명         0
dtype: int64

In [5]:
# 성남시가 아닌 경우
df_credit[~df_credit['시군구명'].str.contains('성남시', na=False)]

,BS_YR_MON,SIGUNGU,BCD,AGE,TOT_CNT,ECON_CNT,NECON_CNT,읍면동명,시군구명


In [6]:
df_credit['AGE'].value_counts()

AGE
20    4979
40    4971
50    4968
60    4968
70    4968
30    4964
Name: count, dtype: int64

In [15]:
df_credit['TOT_CNT'].describe()

count    29818.000000
mean      2195.166778
std       1694.543506
min          0.000000
25%        682.000000
50%       2090.000000
75%       3360.000000
max       7522.000000
Name: TOT_CNT, dtype: float64

In [31]:
df_credit

,BS_YR_MON,SIGUNGU,BCD,AGE,TOT_CNT,ECON_CNT,NECON_CNT,읍면동명,시군구명
0,202301,41131,4113110100,20,2984,2874,66,신흥1동,성남시 수정구
1,202301,41131,4113110100,20,2984,2874,66,신흥2동,성남시 수정구
2,202301,41131,4113110100,20,2984,2874,66,신흥3동,성남시 수정구
3,202301,41131,4113110100,30,3523,3375,105,신흥1동,성남시 수정구
4,202301,41131,4113110100,30,3523,3375,105,신흥2동,성남시 수정구
...,...,...,...,...,...,...,...,...,...
29813,202512,41135,4113511800,30,13,11,0,운중동,성남시 분당구
29814,202512,41135,4113511800,40,0,0,0,운중동,성남시 분당구
29815,202512,41135,4113511800,50,0,0,0,운중동,성남시 분당구
29816,202512,41135,4113511800,60,0,0,0,운중동,성남시 분당구


29818 행 중 504행이 총인구수, 경제활동참여자수, 비경제활동참여수가 0이다.
     - 성남시 분당구 운중동이 486행 나머지는 성남시 수정구 고등동에 분포함

In [6]:
# 읍면동 = 행정동 
df_credit[df_credit['TOT_CNT'] == 0]

,BS_YR_MON,SIGUNGU,BCD,AGE,TOT_CNT,ECON_CNT,NECON_CNT,읍면동명,시군구명
810,202301,41135,4113511700,20,0,0,0,운중동,성남시 분당구
816,202301,41135,4113511800,20,0,0,0,운중동,성남시 분당구
817,202301,41135,4113511800,40,0,0,0,운중동,성남시 분당구
818,202301,41135,4113511800,50,0,0,0,운중동,성남시 분당구
819,202301,41135,4113511800,60,0,0,0,운중동,성남시 분당구
...,...,...,...,...,...,...,...,...,...
29812,202512,41135,4113511800,20,0,0,0,운중동,성남시 분당구
29814,202512,41135,4113511800,40,0,0,0,운중동,성남시 분당구
29815,202512,41135,4113511800,50,0,0,0,운중동,성남시 분당구
29816,202512,41135,4113511800,60,0,0,0,운중동,성남시 분당구


In [9]:
df_credit[
    (df_credit['시군구명'] == '성남시 분당구') &
    (df_credit['읍면동명'] == '운중동') &
    (df_credit['TOT_CNT'] != 0)
]

,BS_YR_MON,SIGUNGU,BCD,AGE,TOT_CNT,ECON_CNT,NECON_CNT,읍면동명,시군구명
780,202301,41135,4113511500,20,1294,1181,74,운중동,성남시 분당구
781,202301,41135,4113511500,30,1306,1238,55,운중동,성남시 분당구
782,202301,41135,4113511500,40,1809,1770,22,운중동,성남시 분당구
783,202301,41135,4113511500,50,1731,1680,35,운중동,성남시 분당구
784,202301,41135,4113511500,60,1219,1171,34,운중동,성남시 분당구
...,...,...,...,...,...,...,...,...,...
29799,202512,41135,4113511700,70,15,12,3,운중동,성남시 분당구
29803,202512,41135,4113511700,50,15,14,0,운중동,성남시 분당구
29804,202512,41135,4113511700,60,15,14,0,운중동,성남시 분당구
29805,202512,41135,4113511700,70,19,13,4,운중동,성남시 분당구


In [18]:
df_credit[
    (df_credit['TOT_CNT'] == 0) &
    (df_credit['ECON_CNT'] == 0) &
    (df_credit['NECON_CNT'] == 0)
][['BS_YR_MON', 'BCD', 'AGE', '읍면동명', '시군구명']]

,BS_YR_MON,BCD,AGE,읍면동명,시군구명
810,202301,4113511700,20,운중동,성남시 분당구
816,202301,4113511800,20,운중동,성남시 분당구
817,202301,4113511800,40,운중동,성남시 분당구
818,202301,4113511800,50,운중동,성남시 분당구
819,202301,4113511800,60,운중동,성남시 분당구
...,...,...,...,...,...
29812,202512,4113511800,20,운중동,성남시 분당구
29814,202512,4113511800,40,운중동,성남시 분당구
29815,202512,4113511800,50,운중동,성남시 분당구
29816,202512,4113511800,60,운중동,성남시 분당구


In [21]:
df42 = df_credit[
    (df_credit['읍면동명'] == '운중동') &
    (df_credit['TOT_CNT'] == 0) &
    (df_credit['ECON_CNT'] == 0) &
    (df_credit['NECON_CNT'] == 0)
]

In [33]:
df_credit[df_credit['BCD'] == 4113511700]

,BS_YR_MON,SIGUNGU,BCD,AGE,TOT_CNT,ECON_CNT,NECON_CNT,읍면동명,시군구명
804,202301,41135,4113511700,20,18,17,0,운중동,성남시 분당구
805,202301,41135,4113511700,30,17,17,0,운중동,성남시 분당구
806,202301,41135,4113511700,40,15,15,0,운중동,성남시 분당구
807,202301,41135,4113511700,50,17,16,0,운중동,성남시 분당구
808,202301,41135,4113511700,60,19,19,0,운중동,성남시 분당구
...,...,...,...,...,...,...,...,...,...
29801,202512,41135,4113511700,30,0,0,0,운중동,성남시 분당구
29802,202512,41135,4113511700,40,0,0,0,운중동,성남시 분당구
29803,202512,41135,4113511700,50,15,14,0,운중동,성남시 분당구
29804,202512,41135,4113511700,60,15,14,0,운중동,성남시 분당구


In [24]:
df42

,BS_YR_MON,SIGUNGU,BCD,AGE,TOT_CNT,ECON_CNT,NECON_CNT,읍면동명,시군구명
810,202301,41135,4113511700,20,0,0,0,운중동,성남시 분당구
816,202301,41135,4113511800,20,0,0,0,운중동,성남시 분당구
817,202301,41135,4113511800,40,0,0,0,운중동,성남시 분당구
818,202301,41135,4113511800,50,0,0,0,운중동,성남시 분당구
819,202301,41135,4113511800,60,0,0,0,운중동,성남시 분당구
...,...,...,...,...,...,...,...,...,...
29812,202512,41135,4113511800,20,0,0,0,운중동,성남시 분당구
29814,202512,41135,4113511800,40,0,0,0,운중동,성남시 분당구
29815,202512,41135,4113511800,50,0,0,0,운중동,성남시 분당구
29816,202512,41135,4113511800,60,0,0,0,운중동,성남시 분당구


In [23]:
df42['AGE'].value_counts()

AGE
20    103
40     92
50     78
70     78
60     72
30     63
Name: count, dtype: int64

성남시 분당구 운중동이 486행 나머지는 성남시 수정구 고등동에 분포함


In [20]:
df_credit[
    (df_credit['읍면동명'] != '운중동') &
    (df_credit['TOT_CNT'] == 0) &
    (df_credit['ECON_CNT'] == 0) &
    (df_credit['NECON_CNT'] == 0)
]

,BS_YR_MON,SIGUNGU,BCD,AGE,TOT_CNT,ECON_CNT,NECON_CNT,읍면동명,시군구명
6028,202308,41131,4113111400,20,0,0,0,고등동,성남시 수정구
6857,202309,41131,4113111400,20,0,0,0,고등동,성남시 수정구
7686,202310,41131,4113111400,20,0,0,0,고등동,성남시 수정구
8515,202311,41131,4113111400,20,0,0,0,고등동,성남시 수정구
9344,202312,41131,4113111400,20,0,0,0,고등동,성남시 수정구
10173,202401,41131,4113111400,20,0,0,0,고등동,성남시 수정구
11002,202402,41131,4113111400,20,0,0,0,고등동,성남시 수정구
11831,202403,41131,4113111400,20,0,0,0,고등동,성남시 수정구
12660,202404,41131,4113111400,20,0,0,0,고등동,성남시 수정구
13489,202405,41131,4113111400,20,0,0,0,고등동,성남시 수정구


## 전입 통계 전처리 / 이상치

In [2]:
file_path = r"C:\Users\cute3\OneDrive\Desktop\pro\final_project\Data\전입통계.csv"

# 파일 존재 여부 먼저 확인
print(os.path.exists(file_path))

# True가 나오면 실행
df_in = pd.read_csv(file_path, encoding="utf-8-sig")

df_in.head(2)

True


,BF_YR_MON,BF_SIDO,BF_SIGNGU,BS_YR_MON,BS_SIGNGU,BS_HCD,TOT_CNT,HIGH_CNT,AVG_INC,AVG_INC_1,...,CNT_3,CNT_4,CNT_5,CNT_6,CNT_7,CNT_8,CNT_9,CNT_10,읍면동명,시군구명
0,202201,41,41131,202301,41111,4111156000,0,0,0,0,...,0,0,0,0,0,0,0,0,파장동,수원시 장안구
1,202201,41,41131,202301,41111,4111156000,0,0,0,0,...,0,0,0,0,0,0,0,0,파장동,수원시 장안구


In [4]:
df_in.shape

(405746, 31)

In [5]:
df_in.isnull().sum()

BF_YR_MON     0
BF_SIDO       0
BF_SIGNGU     0
BS_YR_MON     0
BS_SIGNGU     0
BS_HCD        0
TOT_CNT       0
HIGH_CNT      0
AVG_INC       0
AVG_INC_1     0
AVG_INC_2     0
AVG_INC_3     0
AVG_INC_4     0
AVG_INC_5     0
AVG_INC_6     0
AVG_INC_7     0
AVG_INC_8     0
AVG_INC_9     0
AVG_INC_10    0
CNT_1         0
CNT_2         0
CNT_3         0
CNT_4         0
CNT_5         0
CNT_6         0
CNT_7         0
CNT_8         0
CNT_9         0
CNT_10        0
읍면동명          0
시군구명          0
dtype: int64

## 전출 통계 이상치 / 결측치

In [6]:
file_path = r"C:\Users\cute3\OneDrive\Desktop\pro\final_project\Data\전출통계.csv"

# 파일 존재 여부 먼저 확인
print(os.path.exists(file_path))

# True가 나오면 실행
df_out = pd.read_csv(file_path, encoding="utf-8-sig")

df_out.head(2)

True


,BS_YR_MON,BS_SIGNGU,BS_HCD,AF_YR_MON,AF_SIDO,AF_SIGNGU,TOT_CNT,HIGH_CNT,AVG_INC,AVG_INC_1,...,CNT_3,CNT_4,CNT_5,CNT_6,CNT_7,CNT_8,CNT_9,CNT_10,읍면동명,시군구명
0,202201,41111,4111156000,202301,41,41131,0,0,0,0,...,0,0,0,0,0,0,0,0,파장동,수원시 장안구
1,202201,41111,4111156000,202301,41,41131,0,0,0,0,...,0,0,0,0,0,0,0,0,파장동,수원시 장안구


In [7]:
df_out.shape

(402335, 31)

In [8]:
df_out.isnull().sum()

BS_YR_MON     0
BS_SIGNGU     0
BS_HCD        0
AF_YR_MON     0
AF_SIDO       0
AF_SIGNGU     0
TOT_CNT       0
HIGH_CNT      0
AVG_INC       0
AVG_INC_1     0
AVG_INC_2     0
AVG_INC_3     0
AVG_INC_4     0
AVG_INC_5     0
AVG_INC_6     0
AVG_INC_7     0
AVG_INC_8     0
AVG_INC_9     0
AVG_INC_10    0
CNT_1         0
CNT_2         0
CNT_3         0
CNT_4         0
CNT_5         0
CNT_6         0
CNT_7         0
CNT_8         0
CNT_9         0
CNT_10        0
읍면동명          0
시군구명          0
dtype: int64